Xlib/ext/nvcontrol.py

```diff
@@ -0,0 +1,5358 @@
+
+
+def get_cuda_cores(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_GPU_CORES)
+
+
+def get_memory_bus_width(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_GPU_MEMORY_BUS_WIDTH)
+
+
+def get_total_dedicated_gpu_memory(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_TOTAL_DEDICATED_GPU_MEMORY)
+
+
+def get_used_dedicated_gpu_memory(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_USED_DEDICATED_GPU_MEMORY)
+
+
+def get_pcie_current_link_width(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_GPU_PCIE_CURRENT_LINK_WIDTH)
+
+
+def get_pcie_max_link_width(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_GPU_PCIE_MAX_LINK_WIDTH)
+
+
+def get_pcie_generation(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_GPU_PCIE_GENERATION)
+
+
+def get_video_encoder_utilization(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_VIDEO_ENCODER_UTILIZATION)
+
+
+def get_video_decoder_utilization(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_VIDEO_DECODER_UTILIZATION)
+
+
+def get_current_performance_level(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_GPU_CURRENT_PERFORMANCE_LEVEL)
+
+
+def get_gpu_nvclock_offset(self, target, perf_level):
+    return query_int_attribute(self, target, perf_level, NV_CTRL_GPU_NVCLOCK_OFFSET)
+
+
+def set_gpu_nvclock_offset(self, target, perf_level, offset):
+    return set_int_attribute(self, target, perf_level, NV_CTRL_GPU_NVCLOCK_OFFSET, offset)
+
+
+def get_gpu_nvclock_offset_range(self, target):
+    return query_valid_attr_values(self, target, 0, NV_CTRL_GPU_NVCLOCK_OFFSET)
+
+
+def get_mem_transfer_rate_offset(self, target, perf_level):
+    return query_int_attribute(self, target, perf_level, NV_CTRL_GPU_MEM_TRANSFER_RATE_OFFSET)
+
+
+def set_mem_transfer_rate_offset(self, target, perf_level, offset):
+    return set_int_attribute(self, target, perf_level, NV_CTRL_GPU_MEM_TRANSFER_RATE_OFFSET, offset)
+
+
+def get_mem_transfer_rate_offset_range(self, target):
+    return query_valid_attr_values(self, target, 0, NV_CTRL_GPU_MEM_TRANSFER_RATE_OFFSET)
+
+
+def get_cooler_manual_control_enabled(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_GPU_COOLER_MANUAL_CONTROL) == 1
+
+
+def set_cooler_manual_control_enabled(self, target, enabled):
+    return set_int_attribute(self, target, 0, NV_CTRL_GPU_COOLER_MANUAL_CONTROL, 1 if enabled else 0) == 1
+
+
+def get_fan_duty(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_THERMAL_COOLER_CURRENT_LEVEL)
+
+
+def get_fan_rpm(self, target):
+    return query_int_attribute(self, target, 0, NV_CTRL_THERMAL_COOLER_SPEED)
+
+
+def get_max_displays(self, target):
+    """return the maximum number of display devices that can be driven
+    simultaneously on a GPU (e.g., that can be used in a MetaMode at once).
```

Comment:
It would be nice if all doc strings were complaint with [PEP-0257](https://www.python.org/dev/peps/pep-0257/).

Reply:
Sorry I'm new to Python so I may ask silly questions: what exactly is the problem with this doc string?

Reply:
This is about multi-line doc strings. Quote:

> Multi-line docstrings consist of a summary line just like a one-line docstring, followed by a blank line, followed by a more elaborate description.